### PR TITLE
Fixing vulnerabilities for `source-salesforce` - Premium support

### PR DESCRIPTION
**Pull request by mauricioalarcon**
CVE-2022-40897
https://security-tracker.debian.org/tracker/CVE-2023-29383 
https://security-tracker.debian.org/tracker/CVE-2023-31484 
https://security-tracker.debian.org/tracker/CVE-2016-2781

## What
Changed source base image to use `python:3.9-alpine3.18` instead of `slim` - Adjusted to update OS packages

## How
Switched to use alpine based image

## Recommended reading order
N/A

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*
None expected, just more security 

So far nothing breaks and all the test are passing on our side, I've bumped the minor version accordingly 

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by sajarin** [comment]
@mauricioalarcon can you sign the cla, add an entry to the salesforce.md docs and bump the version in the metadata.yaml file?

File: airbyte-integrations/connectors/source-salesforce/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,38 @@
...
+
+# add default timezone settings
+ENV TZ UTC
+RUN cp /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
 
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=2.1.0
+LABEL io.airbyte.version=2.1.1
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-salesforce/Dockerfile:37]: Can you bump the metadata and documentation changelog? I'll trigger the tests to validate the change.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
Running tests in https://github.com/airbytehq/airbyte/pull/27954

**Comment by mauricioalarcon** [comment]
Sorry guys @marcosmarxm , @sajarin   - while trying to add my email to have the CLA signed ended bringing some other changes - I'm closing this one in favor of https://github.com/airbytehq/airbyte/pull/28021 with the same change.

**Pull request closed without merging**